common/test/test_migrations/utils.py

```diff
@@ -0,0 +1,65 @@
+"""
+General class for Test migrations.For more information visit at
+https://www.caktusgroup.com/blog/2016/02/02/writing-unit-tests-django-migrations/
+"""
+# pylint: disable=redefined-outer-name
+
+from django.test import TransactionTestCase
+from django.db.migrations.executor import MigrationExecutor
+from django.db import connection
+
+
+class TestMigrationsForward(TransactionTestCase):
+    """ Base class for testing forward migrations. """
+    migrate_from = None
+    migrate_to = None
+    app = None
+
+    def setUp(self, execute_forward=True):
+        super(TestMigrationsForward, self).setUp()
+        if execute_forward:
+            self.checkData(self.migrate_from)
+            self.execute_migration(self.migrate_from, self.migrate_to)
+
+    def execute_migration(self, previous, next):
+        """
+        Execute migration from state to another.
+        """
+        # Reverse to the original migration
+        self.executor.migrate(previous)
+
+        self.setUpBeforeMigration(self.old_apps)
+
+        # Run the migration to test
+        self.executor.migrate(next)
+
+        self.apps = self.executor.loader.project_state(next).apps
+
+    def checkData(self, migration_state):
+        """
+        Migrate_from, migrate_to and app variables must be define first.
+        """
+        assert self.migrate_from and self.migrate_to, \
+            "TestCase '{}' must define migrate_from and migrate_from properties".format(type(self).__name__)
+        assert self.app, "app must be define in the TestCase"
+        self.migrate_from = [(self.app, self.migrate_from)]
+        self.migrate_to = [(self.app, self.migrate_to)]
+        self.executor = MigrationExecutor(connection)
+        self.old_apps = self.executor.loader.project_state([(self.app, migration_state)]).apps
+
+    def setUpBeforeMigration(self, apps):  # pylint: disable=invalid-name
```

Comment:
the purpose of apps argument to get_model from apps example `BlogPost = apps.get_model('blog', 'Post')`  but now this can also be done using directly import the model example `from example.app.models import Post`